src/utils/calendar.ts

```diff
@@ -0,0 +1,44 @@
+import ical from "ical-generator";
+
+import { VenueEvent } from "types/venues";
+
+import { eventEndTime, eventStartTime } from "utils/event";
+
+import { WithVenueId } from "utils/id";
+
+import { getFullVenueInsideUrl } from "utils/url";
+
+export interface CreateCalendarProps {
+  calendarName: string;
+  events: WithVenueId<VenueEvent>[];
+}
+
+const createCalendar = ({ calendarName, events }: CreateCalendarProps) => {
+  const cal = ical({ name: calendarName });
+
+  events.forEach((event) =>
+    cal.createEvent({
+      start: eventStartTime(event),
+      end: eventEndTime(event),
+      organizer: `${event.host} <undefined>`, // string format: "name <email>". email cannot be blank
+      description: event.description,
+      summary: event.name,
+      url: getFullVenueInsideUrl(event.venueId),
+    })
+  );
+  return cal;
+};
+
+export const downloadCalendar = ({
+  calendarName,
+  events,
+}: CreateCalendarProps) => {
+  const cal = createCalendar({ calendarName: calendarName, events: events });
+  const outputCal = cal.toURL();
```

Comment:
Not sure if we will need to explicitly release this from memory once we are done with it, or if the lib will handle that for us:

- https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL#memory_management
- https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL

Reply:
This was closed, but with no reason/context given one way or the other as to why?

Reply:
See L47 in src/utils/calendar.ts d339d9a